apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,31 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
+            if (IncompleteClientAndTooManyPendingTasks())
+            {
+                return Task.FromResult(false);
+            }
+
             ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            return _clientTask.ContinueWith(clientTask =>
+            {
+                return clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            });
         }
 
         /// <inheritdoc />
         public override void Log(ExceptionLoggerContext context)
         {
+            if (IncompleteClientAndTooManyPendingTasks())
+            {
+                return;
+            }
+
             ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            _client.ReportErrorEvent(_projectResourceName, errorEvent);
+            _clientTask.ContinueWith(clientTask => 
```

Comment:
No need for a statement lambda here:

```csharp
_clientTask.ContinueWith(task => task.Result.ReportErrorEvent(_projectResourceName, errorEvent);
```

... but you should work out what to do if the task has failed. Currently calling `Result` will just throw an exception...

If the task fails, should we just drop everything on the floor, forever? Or should we throw when trying to log to it?

Additionally, I wouldn't use `ContinueWith` when we know it's completed. That may well end up creating tasks running on different threads, for no particularly good reason.

I think we want logic of:

- Is taskClient completed?
  - No: Have we built up too many tasks?
    - Yes: Ignore
    - No: queue a task to fire if the task completes successfully
  - Yes: Was it successful?
    - Yes: Log (async or sync, accordingly)
    - No: ??? (TBD)